src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -42,9 +42,9 @@ public function __construct(ReflectionClass $reflectedClass)
             $className = $property->getDeclaringClass()->getName();
 
             if ($property->isPrivate() || $property->isProtected()) {
-                $this->hiddenPropertyMap[$className][] = $property->getName();
+                $this->hiddenPropertyMap[$className][] = ObjectProperty::fromReflection($property);
             } else {
-                $this->visiblePropertyMap[] = $property->getName();
+                $this->visiblePropertyMap[] = ObjectProperty::fromReflection($property);
```

Comment:
To be documented similarly to `hiddenPropertyMap`

Reply:
OK

Reply:
Is `array<ObjectProperty>` or `ObjectProperty[]` preferred ?

Reply:
`array<keytype, valuetype>`, since you want to specify the shape of keys

Reply:
`visiblePropertyMap` has no key type, is a list.

Reply:
Should be `array<int, ObjectProperty>`, or even better `list<ObjectProperty>` (the `list` type ensures the first key is `0`, and subsequent keys are adjacent)

Reply:
OK